zap/src/main/java/org/zaproxy/zap/extension/ascan/CustomScanDialog.java

```diff
@@ -642,6 +651,49 @@ private JPanel getTechPanel() {
         return techPanel;
     }
 
+    private void addTagsInFilterPanel(SiteNode siteNode, Collection<String> tags) {
+        if (siteNode != null) {
+            if (siteNode.getHistoryReference() != null
+                    && siteNode.getHistoryReference().getTags() != null) {
+                tags.addAll(siteNode.getHistoryReference().getTags());
+            }
+            Enumeration<?> childEnumeration = siteNode.children();
+            while (childEnumeration.hasMoreElements()) {
+                Object node = childEnumeration.nextElement();
+                if (node instanceof SiteNode) {
+                    addTagsInFilterPanel((SiteNode) node, tags);
+                }
+            }
+        }
+    }
+
+    private JPanel getFilterJPanel(Target target) {
+        if (this.filterPanel == null) {
+            filterPanel = new FilterPanel();
+        } else {
+            filterPanel.resetFilterPanel();
+        }
+
+        Set<String> tags = new LinkedHashSet<>();
```

Comment:
Lets make this responsibility of `FilterPanel`.

Reply:
i have done the changes.